api/envoy/config/route/v3/route_components.proto

```diff
@@ -2391,6 +2392,14 @@ message InternalRedirectPolicy {
   // Allow internal redirect to follow a target URI with a different scheme than the value of
   // x-forwarded-proto. The default is false.
   bool allow_cross_scheme_redirect = 4;
+
+  // Specifies a list of headers, by name, to copy from the internal redirect into the subsequent
+  // request. If a header is specified here but not present in the redirect, it will be cleared in
```

Comment:
Hm, I can defer to the  API shepherds, but I'd find it unexpected to say "preserve foo" and if foo was not sent, have it erase an existing foo header.

@mattklein123 saw you signed off but can I get your thoughts or @htuch here?

Reply:
Yeah that's a good point. Perhaps call is response_headers_to_copy? I think we use that name elsewhere and might better cover what happens.

Reply:
That works for me.